### PR TITLE
Building docker images during CI/CD and publish on tag

### PR DESCRIPTION
**Pull request by edmondop**
Solves #28

### CODE REVIEW

**Review by austin362667 — APPROVED** [review]
Looks good to me. Thank you @edmondop !

File: .github/workflows/k8s.yml [hunk 1]
```diff
@@ -0,0 +1,32 @@
+name: Kubernetes
```
Comment (andygrove) [inline .github/workflows/k8s.yml:1]: Could you add the ASF license header? It looks like we don't have the RAT checks set up in CI yet.

Comment (andygrove) [reply .github/workflows/k8s.yml:1]: I would also be okay with explicitly excluding GitHub workflows from RAT checks if we had RAT set up, so maybe this is fine.

Comment (andygrove) [reply .github/workflows/k8s.yml:1]: I'll go ahead and merge so we can test. I added a bullet to https://github.com/apache/datafusion-ray/issues/2 for setting up RAT.

**Pull request merged**